jormungandr-lib/src/interfaces/fragment_log_persistent.rs

```diff
@@ -14,3 +30,89 @@ pub struct PersistentFragmentLog {
     #[serde(with = "FragmentDef")]
     pub fragment: Fragment,
 }
+
+pub struct FileFragments {
+    reader: BufReader<fs::File>,
+    file_path: PathBuf,
+}
+
+pub struct FileFragmentsIterator {
+    reader: BufReader<fs::File>,
+    file_path: PathBuf,
+    counter: usize,
+}
+
+impl FileFragments {
+    pub fn from_path(file_path: PathBuf) -> std::io::Result<Self> {
+        fs::File::open(file_path.clone()).map(|file| Self {
+            reader: BufReader::new(file),
+            file_path,
+        })
+    }
+}
+
+impl IntoIterator for FileFragments {
+    type Item = Result<PersistentFragmentLog, Error>;
+    type IntoIter = FileFragmentsIterator;
+
+    fn into_iter(self) -> Self::IntoIter {
+        FileFragmentsIterator {
+            reader: self.reader,
+            file_path: self.file_path,
+            counter: 0,
+        }
+    }
+}
+
+impl Iterator for FileFragmentsIterator {
+    type Item = Result<PersistentFragmentLog, Error>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        // EOF reached when buffer is empty after reading successfully at least one.
+        // Then we stop the iterator.
+        if self.reader.buffer().is_empty() && self.counter != 0 {
+            return None;
+        }
+        let codec = bincode::DefaultOptions::new()
+            .with_fixint_encoding()
+            .allow_trailing_bytes();
+        let current = self.counter;
+        self.counter += 1;
+        Some(
```

Comment:
nit:

```
let result = ...;
Some(result)
```

would just look better

Reply:
I had it like that in a previous version though.